src/MVector.jl

```diff
@@ -18,6 +18,9 @@ end
 let dimension_names = QuoteNode.([:x, :y, :z, :w])
     body = :(getfield(v, name))
     for (i,dim_name) in enumerate(dimension_names)
+        @eval @inline Base.propertynames(v::Union{SVector{$i},MVector{$i}}, private::Bool = false) =
+            private ? ($(first(dimension_names, i)...), :data) : ($(first(dimension_names, i)...),)
```

Comment:
FWIW IMO it's nicer to deduplicate like this:

```suggestion
        @eval @inline function Base.propertynames(v::Union{SVector{$i},MVector{$i}}, private::Bool = false)
            named_dims = $(first(dimension_names, i)...)
            private ? (named_dims..., :data) : named_dims
        end
```